common-content/en/module/decomposition/backend-statefulness/index.md

```diff
@@ -0,0 +1,89 @@
++++
+title = "Backend statefulness"
+headless = true
+time = 30
+facilitation = false
+emoji= "📖"
+objectives = [
+    "Identify whether some program is stateful or stateless.",
+]
++++
+
+Our example backend is **stateless**. If you make several requests to it, it will always do the same thing (even though doesn't always return exactly the same result!).
+
+If you made a request from a different computer, or from different country, or on a different day, it would keep doing exactly the same thing. If you restarted the server, it would keep doing exactly the same thing.
+
+If our backend allowed users to add new quotes, it would start having _state_. It would need to remember what quotes had been added. It would be **stateful**.
+
+```js
+import express from "express";
+
+const app = express();
```

Comment:
Given that we haven't taught them express, do you think we should? I don't have. strong view on this just making sure you've thought about it.

Reply:
I don't think so - I think let's just treat this like a library they can pick up like any other, and not make it seem like a big deal or like it should be hard/complicated.